packit/api.py

```diff
@@ -732,6 +734,43 @@ def sync_status_string(
 """
         return f"'{source_git}' is up to date with '{dist_git}'."
 
+    def check_version_distance(
+        self,
+        current,
+        proposed,
+        target_branch,
+    ) -> None:
+        """Following this guidelines:
+        https://docs.fedoraproject.org/en-US/fesco/Updates_Policy/#philosophy
+        we want to avoid major updates of packages within a **stable** release.
+
+        current: str, already released version for package
+        proposed: str, release we are preparing for package
+        target_branch: str, Fedora branch where release the package
+        """
+        if target_branch in get_branches("fedora-branched"):
```

Comment:
related to my other comment, should we leave this to the user to configure or enforce it ourselves as the current implementation does?

Reply:
We can make it customizable, let me know if you think we need it. Here I am trying to enforce what the guidelines say; in rawhide a packager can release what he wants in stable branches there should be no major updates.

Reply:
personally I would probably leave it up to the user, they can configure the jobs per branch and use different masks as they want and I would emphasize the guidelines recommendation in the docs for this option

Reply:
I can see you have introduced `version_update_mask_targets`, I am not against that, but what I meant was that a user could configure multiple jobs, e.g.:

```yaml
jobs:
- job: pull_from_upstream
  trigger: release
  version_update_mask: \d+\.\d+\.
  dist_git_branches:
  - fedora-branched
  
- job: pull_from_upstream
  trigger: release
  dist_git_branches:
    - fedora-rawhide
```   
and if `version_update_mask` is not configured, we would not enforce anything at all

@lachmanfrantisek @majamassarini what do you think is better from the user POV?

Reply:
oh now I follow you and I think it was already that way, I got confused here...
The above code `if target_branch in get_branches("fedora-branched")` is to *always* exclude `rawhide` or the `main` branch for the version check (even when the mask is used on top of the configuration).
Otherwise the user will always need to specify a separate job for rawhide because it does not make much sense to avoid new versions in rawhide... but if we want this then we can simply skip the check.
In both cases I don't know, at this point, if we want to keep the `version_update_mask_targets`?

Reply:
I would probably avoid any internal logic about differentiating branches -- let's behave the same and let the user define multiple jobs if needed.

Reply:
I have removed `version_update_mask_targets` and the check for rawhide so PTAL 🙏🏻🙏🏻🙏🏻

> The nice thing on this feature is, that you can share the mask but have different versions in different branches (e.g. Rawhide updates 2.x and stable Fedoras updates 1.x.)

Honestly I don't totally agree, I started working on this asking why not using `upstream_exclude_tag` or `upstream_include_tag` and as I got it using the `update_version_mask` is handy when you can write it once and forget about it without the need of tweaking it anymore. 
Your example works if someone has already changed the Version in the specfile for rawhide to 2.x. This time you need to tweak the specfile (instead of the .packit.yaml file) manually.
For my simple packages I don't want to manually change things in the rawhide specfile when a new major version is released but still I want be safe and I don't want a new major release in branched branches so I will always create this two jobs:

```
jobs:
- job: pull_from_upstream
  trigger: release
  version_update_mask: \d+\.\d+\.
  dist_git_branches:
  - fedora-branched
  
- job: pull_from_upstream
  trigger: release
  dist_git_branches:
    - fedora-rawhide
```

but I am not a "real" maintainer and maybe I am missing something else  😅